ktor-network/common/test/io/ktor/network/sockets/tests/UDPSocketTest.kt

```diff
@@ -1,42 +1,29 @@
 /*
-* Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
-*/
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
 
 package io.ktor.network.sockets.tests
 
 import io.ktor.network.selector.*
 import io.ktor.network.sockets.*
+import io.ktor.test.dispatcher.*
+import io.ktor.util.*
 import io.ktor.util.network.*
 import io.ktor.utils.io.core.*
+import kotlinx.atomicfu.*
 import kotlinx.coroutines.*
-import kotlinx.coroutines.debug.junit4.*
-import org.junit.*
-import java.net.*
-import kotlin.coroutines.*
-import kotlin.io.use
 import kotlin.test.*
-import kotlin.test.Test
 
-class UDPSocketTest : CoroutineScope {
-    private val testJob = Job()
-    private val selector = ActorSelectorManager(Dispatchers.Default + testJob)
+// TODO: some tests are logging the following exception (tests still green). Not sure why this is happening, this was not the case before rebasing.
+//  io.ktor.utils.io.errors.PosixException.BadFileDescriptorException: EBADF (9): Bad file descriptor
+class UDPSocketTest {
 
-    @get:Rule
-    val timeout = CoroutinesTimeout(1000, cancelOnTimeout = true)
-
-    override val coroutineContext: CoroutineContext
-        get() = testJob
-
-    @AfterTest
-    fun tearDown() {
-        testJob.cancel()
-        selector.close()
-    }
+    private val done = atomic(0)
 
     @Test
-    fun testBroadcastFails(): Unit = runBlocking {
+    fun testBroadcastFails(): Unit = testUdpSockets { selector ->
         if (OS_NAME == "win") {
```

Comment:
Why is the `testBroadcastFails` test skipped on JVM on Windows? Should it be skipped in Kotlin/Native on Windows as well?

Reply:
Windows using Kotlin/Native is not supported by this module.